src/rdiff_backup/rpath.py

```diff
@@ -1167,15 +1167,17 @@ def write_from_fileobj(self, fp, compress=None):
     def write_string(self, s, compress=None):
         """Write string s into rpath"""
         assert not self.lstat(), "File '{rp!r}' already exists".format(rp=self)
-        with self.open("w", compress=compress) as outfp:
-            outfp.write(s)
+        outfp = self.open("w", compress=compress)
+        outfp.write(s)
+        outfp.close()
```

Comment:
How is it related to the locking !?

Reply:
The tests were failing with those lines. I assume that two `with` one within the other wasn't working so well.